pkg/gcp/ocpgwdeployer.go

```diff
@@ -67,11 +79,87 @@ func (d *ocpGatewayDeployer) Deploy(input api.GatewayDeployInput, reporter api.R
 	reporter.Succeeded("Opened External ports %q with firewall rule %q on GCP",
 		formatPorts(input.PublicPorts), externalIngress.Name)
 
+	numGatewayNodes, eligibleZonesForGW, err := d.parseCurrentGatewayInstances(reporter)
+	if err != nil {
+		reporter.Failed(err)
+		return err
+	}
+
+	if numGatewayNodes == input.Gateways {
+		reporter.Succeeded(messageValidatedCurrentGWs)
+		return nil
+	}
+
+	// Currently, we only support increasing the number of Gateway nodes which could be a valid use-case
+	// to convert a non-HA deployment to an HA deployment. We are not supporting decreasing the Gateway
+	// nodes (for now) as it might impact the datapath if we accidentally delete the active GW node.
+	if numGatewayNodes < input.Gateways {
+		gatewayNodesToDeploy := input.Gateways - numGatewayNodes
+
+		if d.dedicatedGWNode {
+			for _, zone := range eligibleZonesForGW.Elements() {
+				reporter.Started(fmt.Sprintf("Deploying dedicated gateway node in zone %q", zone))
```

Comment:
nit but we're using a mix of "gateway" and "Gateway" in the messages. I think we should be consistent. I prefer "gateway".

Reply:
Done